pulp_ansible/app/serializers.py

```diff
@@ -515,13 +515,33 @@ class CollectionVersionSignatureSerializer(NoArtifactContentSerializer):
     signing_service = RelatedField(
         help_text=_("The signing service used to create the signature."),
         view_name="signing-services-detail",
-        queryset=SigningService.objects.all(),
+        read_only=True,
         allow_null=True,
     )
 
+    def __init__(self, *args, **kwargs):
+        """Ensure `file` field is required."""
+        super().__init__(*args, **kwargs)
+        self.fields["file"].required = True
```

Comment:
I know this "works", but can you check in the api schema that it is really marked as required. If not, i think you need to redefine the field on this class.